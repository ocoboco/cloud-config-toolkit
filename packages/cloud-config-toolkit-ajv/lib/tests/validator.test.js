const Ajv = require('ajv');

const Validator = require('../validator');
const keywords = require('./fixtures/keywords');
const schema = require('./fixtures/schema');
const validConfig = require('./fixtures/config/valid.json');
const invalidConfig = require('./fixtures/config/invalid.json');

describe('Validator', function() {
  test('validates valid configuration', function() {
    const validator = new Validator({
      schema,
      keywords
    });
    expect(validator.isValid(validConfig)).toBeTruthy();
    expect(validator.getErrors(validConfig)).toHaveLength(0);
  });

  test('invalidates invalid configuration', function() {
    const validator = new Validator({
      schema,
      keywords
    });
    expect(validator.isValid(invalidConfig)).toBeFalsy();
    expect(validator.getErrors(invalidConfig)).toMatchSnapshot();
  });
});