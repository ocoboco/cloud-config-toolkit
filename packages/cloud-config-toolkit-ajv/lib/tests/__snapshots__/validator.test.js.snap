// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Validator invalidates invalid configuration 1`] = `
Array [
  Object {
    "dataPath": ".settings",
    "keyword": "required",
    "message": "should have required property 'hostname'",
    "params": Object {
      "missingProperty": "hostname",
    },
    "schemaPath": "#/properties/settings/required",
  },
]
`;
